wal_stubs_test.go

```diff
@@ -302,6 +309,20 @@ func (ts *testStorage) assertValidMetaState(t *testing.T) {
 				require.True(t, ok)
 				require.Equal(t, seg.BaseIndex, log.Index)
 			}
+
+			// Verify that if it's meant to be sealed in metadata that it actually is
+			// and has an index block. Note that we don't test that the actual segment
+			// sealed status matches meta since that is not always true e.g. just
+			// after an append that caused segment to seal but before the rotate has
+			// updated metadata. The thing we need to ensure is that if metadata says
+			// it's sealed that the actual segment is actually sealed and has an index
+			// block.
+			if !seg.SealTime.IsZero() {
+				sealed, indexStart, err := ts.Sealed()
+				require.NoError(t, err)
+				require.True(t, sealed)
+				require.NotEqual(t, 0, int(indexStart))
```

Comment:
Realised I should also assert that the `seg.IndexStart` matches the `indexStart`. It's a _bit_ of a tautology in the sense that segments read from disk would actually populate `indexStart` from metadata in the first place but in these tests it would actually verify something useful: that the segment we just sealed reported it's index offset consistently and that it was actually updated in the meta data. I'll add that before I merge.